### DISCUSSION TITLE
how to update nested object values when fieldArray update method

### DISCUSSION THREAD

**Discussion by ElyseYip** [react-hook-form/react-hook-form#11555] [Q&A] [answered]
when i use the fieldArray update function to updated the object values of propertyType and logic, it doesn't work, only the value to set as "" works, how to resolve this?
update(index, {
        ...item,
        propertyType: {
          propertyType: "",
          propertyTypeName: "",
          propertyTypeId: "",
          type: "",
        },
        logic: {
          operatorName: "",
          operatorId: "",
        },
        value: "",
      });

**Comment by sujinleeme** [comment, marked as answer]
Perhaps the field name you provided may be incorrect.  In this  [codesandbox](https://codesandbox.io/p/sandbox/update-nested-field-array-with-update-6yhgcd?file=%2Fsrc%2Findex.tsx%3A52%2C75), You can check `update()` in the nested field array with the form data that you used. I haven't faced the issue of updating an array with new values.

**Reply by ElyseYip** [reply]
for the value with a primitive string, it has no issue.
but updating the object values is not working , I wonder how can i update the object values of : 
propertyType: {
propertyType: "",
propertyTypeName: "",
propertyTypeId: "",
type: "",
},
logic: {
operatorName: "",
operatorId: "",
},

**Reply by sujinleeme** [reply]
It should work from my side if the field is an array correctly. Maybe could you provide the codesandbox with a minimal code?

**Reply by ElyseYip** [reply]
@sujinleeme i get it solved, as I have to handle more than one update in the onChange function, the sequence of the update make the code i showed invalid! There is no issue of updating the object values using react hook form!
Thanks so much for helping!
